### ISSUE TITLE
Allow wildcards in languages.toml `roots`

### ISSUE THREAD

**Issue by wilcooo** [helix-editor/helix#7419]
```toml
[[language]]
name = "rust"
roots = ["Cargo.toml", "Cargo.lock"]
```

This works fine for projects like rust packages, which always have a `Cargo.toml` in the root.
**However, other projects, like cabal packages (Haskell), only have a file in the root that includes the package name, like `myproject.cabal`.**

Suggestion: allow the `roots` setting to accept wildcards, e.g. `roots = ["*.cabal"]`.

--- 

Note, as long as you are working in a git repo, this is not a problem (I guess that Helix always sees the `.git` folder as a root). However, I'm currently working in a git repo which has a cabal project in a subfolder.

### Temporary fix

For anyone that comes across this issue, create an empty file named `.root`, and add `".root"` to the `roots` setting of the language that you are currently using.

**Comment by kirawi** [comment]
I think that a `.helix` folder will become the top-level root.

**Comment by wilcooo** [comment]
> I think that a `.helix` folder will become the top-level root after `.git` if you add it.

Yes, it does, as does adding a `.root` file after specifying that in languages.toml.

However, it would still be helpful if this would work without manually adding files or folders. It took me a while to figure out that this was the cause for my LSP not working (I got unhelpful LSP errors, while the LSP worked fine when opening the same folder in other editors).

**Comment by pascalkuthe** [comment]
Simply treating all cabal files as root markers is very fragile and not  good solution.  Cabal does have a project file https://cabal.readthedocs.io/en/3.10/cabal-project.html. If you're not using the standard build infrastructure then there is really no reasonable assumption we can make about the project layout. Instead, you can manually provide a list of lsp roots by setting the 'workspace-lsp-roots` option in your workspace config

**Comment by wilcooo** [comment]
The `cabal.project` file is meant to specify settings that apply to multiple packages, which is not applicable here.
A "standard" package (as set up using `cabal init`) really only has a cabal file that could mark the root.
I'm not sure why treating cabal files as root markers is fragile, when would there be cabal files outside of the root of the package?
